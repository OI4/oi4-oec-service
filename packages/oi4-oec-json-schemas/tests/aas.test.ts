import schema from '../src/schemas/AAS.schema.json'

import validAASObjs from './__fixtures__/aas_valid.json'
import invalidAASObjs from './__fixtures__/aas_invalid.json'
import int32 from '../src/schemas/dataTypes/int32.schema.json'
import {matchersWithOptions} from 'jest-json-schema';

expect.extend(
    matchersWithOptions({
        verbose: true,
    }, (ajv) => {
        ajv.addSchema(int32)
    })
)

describe('aas schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it.each(validAASObjs as [])(
    '(%#) match valid config object to schema -> %s',
    (_name: string, obj) => {
      expect(obj).toMatchSchema(schema)
    }
  )

  it.each(invalidAASObjs)(
    '(%#) match fails for invalid config -> %s',
    (name: string, obj) => {
      expect(obj).not.toMatchSchema(schema)
      expect(() => {
        expect(obj).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(name)
    }
  )
})
