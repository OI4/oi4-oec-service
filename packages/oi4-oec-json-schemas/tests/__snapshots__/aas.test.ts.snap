// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aas schema (0) match fails for invalid config -> empty: empty 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'AASId, GlobalAssetId'
    Path:     AAS.schema.json#/required
"
`;

exports[`aas schema (1) match fails for invalid config -> AASId is missing: AASId is missing 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'AASId, GlobalAssetId'
    Path:     AAS.schema.json#/required
"
`;

exports[`aas schema (2) match fails for invalid config -> GlobalAssetId is missing: GlobalAssetId is missing 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'AASId, GlobalAssetId'
    Path:     AAS.schema.json#/required
"
`;

exports[`aas schema (3) match fails for invalid config -> wrong AASId type: wrong AASId type 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'AASId, GlobalAssetId'
    Path:     AAS.schema.json#/required
  /GlobalAssetId must be string
    Received: <number> 50
    Path:     AAS.schema.json#/properties/GlobalAssetId/type
"
`;

exports[`aas schema (4) match fails for invalid config -> wrong GlobalAssetId type: wrong GlobalAssetId type 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'AASId, GlobalAssetId'
    Path:     AAS.schema.json#/required
  /AASId must be string
    Received: <number> 20
    Path:     AAS.schema.json#/properties/AASId/type
"
`;
